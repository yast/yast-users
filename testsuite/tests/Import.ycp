/**
 * File		: Import.ycp
 * Module	: Users configurator
 * Summary	: testing Import function
 * Author	: Jiri Suchomel <jsuchome@suse.cz>
 * $Id$
 */
{
    // testedfiles: Users.pm UsersCache.pm

    import "Users";
    import "UsersCache";
    import "Mode";
    import "Directory";
    import "Progress";

    string tmpdir	= Directory::tmpdir;
    foreach (string file, [ "passwd", "group", "shadow" ], {
	SCR::Execute (.target.bash, sformat ("/bin/cp ./%1 %2/", file, tmpdir));
    });
    Users::SetBaseDirectory (tmpdir);
    Users::ReadLocal ();

    import "Testsuite";

    map READ = $[
	"etc" : $[
	    "fstab": [],
	    "cryptotab": [],
	    "default": $[
		"useradd": $[
		    "home":	"/home",
		    "groups":	"audio,video",
		    "inactive":	nil,
		    "expire":	nil,
		    "shell":	nil,
		    "group":	100
		]
	    ],
	],
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "IO_SCHEDULER"		: "",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
	"target"	: $[
	    "stat"	: $[],
	    "size"	: -1,
	    "string"	: ""
	],
	"sysconfig"	: $[
	    "displaymanager"	: $[
		"DISPLAYMANAGER"	: ""
	    ],
	],
    ];
    map WRITE = $[];
    map EXEC = $[
	"passwd" : $[
	    "init"	: true
	],
	"target" : $[
	    "bash" :	-1,
	    "bash_output" : $[
		"stdout"	: "ggg",
	    ],
	],
    ];

    Mode::SetTest ("test");

    map<string,any> importing = $[
	"users"		: [
	    $[
		"username"	: "ggg",
		"user_password"	: "password"
	    ],
	],
    ];

    Testsuite::Dump ( sformat ("local user names:\n %1", UsersCache::GetUsernames ("local")));

    Testsuite::Test (``(Users::Import (importing)), [READ,WRITE,EXEC], 0);

    Testsuite::Dump ( sformat ("local user names:\n %1", UsersCache::GetUsernames ("local")));

    Testsuite::Test (``(Users::SelectUserByName ("ggg")), [], 0);

    Testsuite::Dump (sformat ("---- user 'ggg':\n %1", Users::GetCurrentUser ()));
}
