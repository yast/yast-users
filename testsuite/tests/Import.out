Dump	local user names:
Dump	 []
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/home"
Read	.etc.default.useradd."inactive" nil
Read	.etc.default.useradd."shell" nil
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Execute	.target.bash_output "echo 'ggg' | grep '^[[:alpha:]_][[:alnum:]_.-]*[[:alnum:]_.$-]\\?$'" $["LANG":"C"] $["stdout":"ggg"]
Read	.target.string "/etc/shells" ""
Return	true
Dump	local user names:
Dump	 ["ggg", "hh", "ii"]
Return	nil
Dump	---- user 'ggg':
Dump	 $["chown_home":true, "cn":"", "create_home":true, "encrypted":true, "gidNumber":"100", "grouplist":$[], "groupname":"users", "homeDirectory":"/home/ggg", "loginShell":"", "modified":"added", "org_homeDirectory":"/home/ggg", "org_uid":"ggg", "org_uidNumber":"1000", "plugins":[], "shadowExpire":"", "shadowFlag":"", "shadowInactive":"", "shadowLastChange":"0", "shadowMax":"99999", "shadowMin":"0", "shadowWarning":"7", "type":"local", "uid":"ggg", "uidNumber":"1000", "userPassword":"crypted_password", "what":"add_user"]
